scheduler/src/cook/kubernetes/api.clj

```diff
@@ -487,9 +511,30 @@
                                                                           .getRequests
                                                                           convert-resource-map))
                                                                 containers)
+                                        _ (log/info "Pod: " pod)
+                                        _ (log/info "Container resources: " (map (fn [^V1Container c]
+                                                                          (some-> c
+                                                                                  .getResources))
+                                                                        containers))
+                                        _ (log/info "Container requests: " (map (fn [^V1Container c]
+                                                                          (some-> c
+                                                                                  .getResources
+                                                                                  .getRequests))
+                                                                        containers))
                                         resource-map (apply merge-with + container-requests)
-                                        gpu-model (some-> pod .getSpec .getNodeSelector (get "cloud.google.com/gke-accelerator"))]
-                                    (add-gpu-model-to-resource-map gpu-model resource-map))))
+                                        _ (log/info "Resource map before adding gpu-model and/or disk-type: " resource-map)
+                                        ^V1NodeSelector nodeSelector (some-> pod .getSpec .getNodeSelector)
+                                        gpu-model (some-> nodeSelector (get "cloud.google.com/gke-accelerator"))
+                                        disk-type (some-> nodeSelector (get (pool->disk-type-label-name pool-name)))]
+                                    (log/info "Resource map after adding gpu-model and/or disk-type: "
+                                              (->> resource-map
+                                                   (force-gpu-model-in-resource-map gpu-model)
+                                                   (force-disk-type-in-resource-map disk-type)))
+                                    (->> resource-map
+                                         (force-gpu-model-in-resource-map gpu-model)
+                                         (force-disk-type-in-resource-map disk-type)))))
+                           ; remove nil resource-maps from collection before deep-merge-with
+                           (remove nil?)
```

Comment:
Would be nice to include in the comment the info about why resource maps can sometimes be nil